### DISCUSSION TITLE
Custom Release titles for GitHub

### DISCUSSION THREAD

**Discussion by KnisterPeter** [intuit/auto#2181] [Q&A] [answered]
Hi there,

first of all this is an awesome tool :rocket:.

I would like to ask if it would be possible to extend it to allow plugins to contribute release titles for GitHub releases.
This https://github.com/intuit/auto/pull/1083 defines that it should be the tag name. But before it was set implicitly which might have had some side-effects. I guess that from the PR message included.

What I have in mind would be a hook for plugins to customize/contribute the release. Or if preferred a configuration option to supply a name.
I would like to have it kind of dynamic, because we are using changing names for each release following a schema (e.g. like ubuntu with this: https://wiki.ubuntu.com/DevelopmentCodeNames)

I'm also willing to contribute a PR if accepted to work on this. But I'm unsure where to start, since I didn't know the codebase and the preferences.

**Comment by hipstersmoothie** [comment, marked as answer]
Yeah I think a hook for that would be fine I'm thinking a hook called `modifyRelease` that gets called before `publish` [here](https://github.com/intuit/auto/blob/43dad64aba18628711efdbea06b5581e2a77e276/packages/core/src/auto.ts#L563-L568).

The hook could get the release notes, tag and title and allow the hook to modify it. You could also then add a tiny plugin that implements the desired functionality.

---

If you didn't want to go this route though I think it would be possible to accomplish this without modifying auto.

1. Create a plugin that taps `makeRelease`
2. I _think_ this should run after the default behavior. You'll have a reference to the tag that was release
3. Use octokit and that tag to modify the release that was just made to follow your naming

**Reply by laughedelic** [reply]
I think the hook proposed in this answer would also solve https://github.com/intuit/auto/discussions/2124
